File: 1-100/26.remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int lastIdx = nums.size();
        if (lastIdx < 2)
        {
            return lastIdx;
        }

        int retIdx = 0;
        int firstIdx = 0;
        while (++firstIdx != lastIdx)
        {
            if (nums[retIdx] != nums[firstIdx])
            {
                nums[++retIdx] = nums[firstIdx];
            }
        }

        return ++retIdx;
    }
};

File: 1-100/26.remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        nLen = len(nums)
        if nLen < 2:
            return nLen
        
        ret = 0
        for x in nums[1:]:
            if nums[ret] != x:
                ret += 1
                nums[ret] = x
        
        return ret + 1
